lib/sink/connection/client/connection_status.ex

```diff
@@ -0,0 +1,72 @@
+defmodule Sink.Connection.Client.ConnectionStatus do
+  @moduledoc """
+  State machine that manages a connection's state and state transitions
+  """
+  alias Sink.Connection.Protocol
+
+  @type t :: %__MODULE__{
+          connection_state: :requesting_connection | :connected | :disconnecting,
+          server_identifier: Protocol.server_identifierentifier() | nil,
+          reason: nil | binary
+        }
+
+  defstruct [
+    :connection_state,
+    :server_identifier,
```

Comment:
How about `instance_id`? That should be generic enough and implies that the identifier is tied to the instance of the pairing of client and server vs just the server.